src/js/contexts/MessageContext/MessageContext.js

```diff
@@ -25,6 +25,7 @@ export const format = (options, messages) => {
   // their bundles that way and don't need to pass the messages
   // themselves in this property, just the format function.
   const idParts = options.id?.split('.') || [];
+  // eslint-disable-next-line no-unsafe-optional-chaining
   const baseId = idParts[idParts?.length - 1];
```

Comment:
Any reason not to do `const baseId = idParts[(idParts?.length || 0) - 1];`

Reply:
I debated this one and am open to persuasion. My thinking was that if we did that, we are then taking a negative index of our array and that becomes undefined which leaves us in pretty much the same situation, taking a NaN index of an array which is also undefined.

I kind of felt like adding the `||` provided a false sense of security without improving the situation. I opted for disabling the linter rule to at least serve as a 'caution flag' if we needed to re-visit this logic.

Reply:
Yeah that makes sense, I agree switching to `(idParts?.length || 0)` does provide a false sense of security and doesn't improve anything besides making lint happy. I'm good with leaving the rule disabled (or if it isn't too complicated maybe changing the logic so that if the idParts array is empty we just return to avoid an index out of bounds issue).